app/app.go

```diff
@@ -84,6 +84,7 @@ func MakeCodec() *codec.Codec {
 
 	ModuleBasics.RegisterCodec(mesgcodec.Codec)
 	vesting.RegisterCodec(mesgcodec.Codec)
+	ownership.RegisterCodec(mesgcodec.Codec)
```

Comment:
this should not be done here.
the call to `ModuleBasics.RegisterCodec(mesgcodec.Codec)` should iterate over all the module and register the codec against each code. The module has to itself on the codec by implementing [func (AppModuleBasic) RegisterCodec(cdc *codec.Codec) {](https://github.com/mesg-foundation/engine/pull/1651/files#diff-a1fb0f761753d7736d7a16cbeed04b6aR34) but you already did it.

Reply:
In your currently implementation, this is required because your don't register the right ownership AppModuleBasic. See comment line #62 of this same file